### PR TITLE
[Test Suite] - Update Test Suite with new modifications

### PR DESCRIPTION
**Pull request by malshaniS**
## [Test Suite] - Update Test Suite with new modifications

> Update Test Suite with new modifications and add new test cases.

**Issue link:** *[#36](https://github.com/wso2-enterprise/solutions-bfsi/issues/36)*

**Doc Issue:** *Optional, link issue from [documentation repository](https://github.com/wso2/docs-ob/issues)*

**Applicable Labels:** *Spec, product, version, type (specify requested labels)*

------

### Development Checklist

1. [x] Built complete solution with pull request in place.
2. [ ] Ran checkstyle plugin with pull request in place.
3. [ ] Ran Findbugs plugin with pull request in place.
4. [ ] Ran FindSecurityBugs plugin and verified report.
5. [x] Formatted code according to WSO2 code style.
6. [ ] Have you verify the PR does't commit any keys, passwords, tokens, usernames, or other secrets?
7. [ ] Migration scripts written (if applicable).
8. [ ] Have you followed secure coding standards in [WSO2 Secure Engineering Guidelines](http://wso2.com/technical-reports/wso2-secure-engineering-guidelines)?

### Testing Checklist

1. [ ] Written unit tests.
2. [x] Documented test scenarios(link available in guides).
3. [ ] Written automation tests (link available in guides).
4. [ ] Verified tests in multiple database environments (if applicable).
5. [ ] Verified tests in multiple deployed specifications (if applicable).
6. [x] Tested with OBBI enabled  (if applicable).
7. [ ] Tested with specification regulatory conformance suites  (if applicable).

**Automation Test Details**

| Test Suite        | Test Script IDs   |
| ----------------- | ----------------- |
| Integration Suite | *TCXXXXX, TCXXXX* |

**Conformance Tests Details**

| Test Suite Name  | Test Suite Version | Scenarios  | Result   |
| ---------------- | ------------------ | ---------- | -------- |
| *Security Suite* | *VX.X*             | *Foo, Bar* | *Passed* |

## Resources

**Knowledge Base:** https://sites.google.com/wso2.com/open-banking/

**Guides:** https://sites.google.com/wso2.com/open-banking/developer-guides

### CODE REVIEW

**Review by anjuchamantha — DISMISSED** [review]

**Review by anjuchamantha — APPROVED** [review]

File: integration-test-suite/cds-toolkit-integration-test/src/test/groovy/org/wso2/cds/integration/test/accounts/AccountsRetrievalRequestHeaderValidationTest.groovy [hunk 1]
```diff
@@ -378,7 +378,7 @@ class AccountsRetrievalRequestHeaderValidationTest extends AUTest {
 
         def response = AURequestBuilder.buildBasicRequestWithCustomHeaders(userAccessToken,
                 AUConstants.X_V_HEADER_ACCOUNTS, clientHeader)
-                .queryParam(AUConstants.PARAM_PRODUCT_CATEGORY, "TRANS")
+                .queryParam(AUConstants.PARAM_PRODUCT_CATEGORY, "TANS")
```
Comment (imesh94) [inline integration-test-suite/cds-toolkit-integration-test/src/test/groovy/org/wso2/cds/integration/test/accounts/AccountsRetrievalRequestHeaderValidationTest.groovy:381]: Is this a typo?

Comment (malshaniS) [reply integration-test-suite/cds-toolkit-integration-test/src/test/groovy/org/wso2/cds/integration/test/accounts/AccountsRetrievalRequestHeaderValidationTest.groovy:381]: No This is a test cases to validate the behaviour when we send an invalid product-category value.

**Comment by anjuchamantha** [comment]
Hi @malshaniS 

The readme in integration-test-suite says to build both [bfsi-test-framework](https://github.com/wso2-enterprise/financial-open-banking/tree/master/integration-test-suite/bfsi-test-framework) and [open-banking-test-framework](https://github.com/wso2-enterprise/financial-open-banking/tree/master/integration-test-suite/open-banking-test-framework). Is this necessary? Those are non public repos, and we cannot say to build those in a public repo right?

Shall we check and modify the readme if needed as well?

Thanks.

**Comment by malshaniS** [comment]
Hi @anjuchamantha,

Updated the links.

File: integration-test-suite/README.md [hunk 2]
```diff
@@ -6,8 +6,8 @@ This test suite covers all the functional scenarios and use cases of CDS Toolkit
 
 1. Clone the master branch of the [financial-open-banking](https://github.com/wso2-enterprise/financial-open-banking/tree/master) repository.
 2. Goto "integration-test-suite" and build the following modules (These are the base modules of open banking test framework);
```
[the remark this answers is not in the document]
Comment (malshaniS) [reply integration-test-suite/README.md]: Modified. Please check the latest commit

**Review by anjuchamantha — APPROVED** [review]

**Review by imesh94 — APPROVED** [review]

**Pull request merged**